osf/models/analytics.py

```diff
@@ -1,25 +1,151 @@
+import logging
+
 from dateutil import parser
-from django.db import models
+from django.db import models, transaction, connection
+from django.utils import timezone
+
+from framework.sessions import session
 from osf.models.base import BaseModel
 from osf.utils.datetime_aware_jsonfield import DateTimeAwareJSONField
 
+logger = logging.getLogger(__name__)
+
 
 class UserActivityCounter(BaseModel):
     _id = models.CharField(max_length=255, null=False, blank=False, db_index=True,
                            unique=True)
     action = DateTimeAwareJSONField(default=dict)
     date = DateTimeAwareJSONField(default=dict)
-    total = models.PositiveIntegerField()
+    total = models.PositiveIntegerField(default=0)
 
     @classmethod
-    def increment(self, user_id, action, date_string):
+    def get_total_activity_count(cls, user_id):
+        try:
+            return cls.objects.get(_id=user_id).total
+        except cls.DoesNotExist:
+            return 0
+
+    @classmethod
+    def increment(cls, user_id, action, date_string):
         date = parser.parse(date_string).strftime('%Y/%m/%d')
-        date = date
+        with transaction.atomic():
+            # select_for_update locks the row but only inside a transaction
+            uac, created = cls.objects.select_for_update().get_or_create(_id=user_id)
```

Comment:
`select_for_update` inside `transaction.atomic` should force a row lock until the transaction is committed, when the context manager is left. This should make our increments/updates isolated and safe.